providers/ms365/resources/ms365_exchange.go

```diff
@@ -184,6 +198,32 @@ func convertTeamsProtectionPolicy(r *mqlMs365Exchangeonline, data []*TeamsProtec
 	return plugin.TValue[[]interface{}]{Data: result, State: plugin.StateIsSet, Error: teamsProtectionPolicyErr}
 }
 
+// Related to ReportSubmissionPolicy as a seperate function
+func convertReportSubmissionPolicy(r *mqlMs365Exchangeonline, data []*ReportSubmissionPolicy) plugin.TValue[[]interface{}] {
```

Comment:
id prefer if we have those func convert to the MQL resource instead. we can leave the setting of  `plugin.TValue` to the caller of this function